src/common/transformations/include/transformations/common_optimizations/PrevSequenceLengthPattern.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/cc/pass/itt.hpp"
+#include "transformations/utils/utils.hpp"
+#include "transformations_visibility.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/shape_of.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API PrevSequenceLengthPattern;
+
+}  // namespace pass
+}  // namespace ov
+
+using namespace ov::op;
+
+class ov::pass::PrevSequenceLengthPattern : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("PrevSequenceLengthPattern", "0");
+    PrevSequenceLengthPattern(const std::shared_ptr<ov::op::v1::Subtract>& prev_max_seq_len) {
+        MATCHER_SCOPE(PrevSequenceLengthPattern);
+ 
+        auto kv_past = pattern::wrap_type<v6::ReadValue>({pattern::any_input()});
+        auto kv_gather = pattern::wrap_type<v8::Gather>({kv_past, pattern::any_input(), pattern::any_input()});
+        auto kv_shape = pattern::wrap_type<v3::ShapeOf>({kv_gather});
+        auto seq = pattern::wrap_type<v8::Gather>({kv_past, pattern::any_input(), pattern::any_input()});
+ 
+        ov::matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](ov::pass::pattern::Matcher& m) {
+        std::cout << "____" << matcher_name << "___Matched___" << std::endl;
+            // TODO: Check that seq has axis that really takes sequence len but not any other dimension -- use symbolics or look at the constant input
+            auto gather = m.get_match_root();
+            auto target_type = gather->get_output_element_type(0);
+            std::shared_ptr<Node> replacement;
+            if (prev_max_seq_len->get_output_element_type(0) != target_type) {
+                std::cout << "Converting " << prev_max_seq_len->get_output_element_type(0) << " of max_context_len to " << target_type << std::endl;
```

Comment:
here it prints `target_type` as `f32`. I wonder that i64 is converted to f32

Reply:
fixed